### PR TITLE
[tinker] Decode vLLM completion bodies straight into numpy with pysimdjson

### PR DESCRIPTION
**Pull request by avigyabb**
Stack 6/7. With results the size of Chuck's repro (262k tokens, 2.8 MB of JSON each), decoding the vLLM body was 72% of the API server's CPU even after 5/7: `orjson.loads` builds a Python object per token and logprob (22 ms), then numpy converts the lists (16 ms).

**Change.** `skyrl/tinker/extra/completion_decode.py`: `CompletionDecoder` parses the body with [pysimdjson](https://github.com/TkTech/pysimdjson) and pulls `token_ids` / `token_logprobs` out as raw numeric buffers (`Array.as_buffer`) into int32/float32 arrays, ~6 ms for a 262k-token result with no per-token Python objects. Falls back to orjson when pysimdjson is not installed, and per array when an array is not purely numeric (a null logprob). Zero-fill semantics for missing/null logprobs are unchanged. `pysimdjson` is added to the `tinker` extra (`uv.lock` +46 lines). Alternatives measured on one 262k body: current 38.6 ms; `np.fromstring` on extracted substrings 28.9 ms; pysimdjson 6.2 ms.

**Results** (proto client, 262k-token results): 512 concurrent 16.6 s -> 6.25 s server CPU, 32 -> 89 results/s. 8192 with 5 s engine queueing and 60 s retrieved TTL: 8192/8192, 0 failures, 79/s (~230 MB/s), 15.9 GB. 131072 x 212k tokens with a 16384 in-flight cap: 130917/131072 in 1144 s at 114/s (~190 MB/s), 0 forwarding errors, 20.4 GB; the 155 misses were `asample` responses over 45 s during stalls, which the SDK retries.

At this size memory is retention: completion rate x 1.7-2.1 MB x `external_future_retrieved_ttl_sec`; set the TTL to 60-120 s for 200k+ token rollouts. In production, vLLM's own JSON serialization of 2-3 MB per result is likely to saturate before this server.

Tests: `tests/tinker/test_completion_decode.py` covers both backends, nulls, empty arrays, prompt-logprob passthrough, non-JSON bodies, parser reuse, and 200k-element parity between the fast and fallback paths.

**Stack** (each PR retargets to `main` as the one below merges)
1. #2160
2. #2161
3. #2162
4. #2163
5. #2164
6. #2165
7. #2166

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes the production vLLM response decode path and RL-facing token/logprob arrays; behavior is guarded by dual-backend tests and preserved zero-fill semantics, but numeric or parsing regressions would affect training rollouts.
> 
> **Overview**
> Speeds up the tinker inference forwarding hot path by **avoiding per-token Python objects** when parsing large vLLM `/v1/completions` responses.
> 
> Adds **`CompletionDecoder`** in `completion_decode.py`, which uses **pysimdjson** to pull `token_ids` and `token_logprobs` straight into **int32/float32 numpy arrays**, with **orjson** as a fallback when simdjson is missing or an array is not purely numeric. Missing/null logprobs are still **zero-filled** so RL shapes stay consistent; `prompt_logprobs` are only materialized when requested.
> 
> **`SkyRLTrainInferenceForwardingClient`** now decodes responses through a reused decoder instead of `orjson.loads`, then builds proto sequences from the typed arrays. **`pysimdjson`** is added to the **`tinker`** extra (and lockfile). Tests cover both backends, edge cases, and large-array parity.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2c3756ad074e3d162399b81dc8b9634c9ee221af. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: skyrl/tinker/extra/completion_decode.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+class DecodedChoice:
+    finish_reason: str | None
+    tokens: np.ndarray  # int32
+    logprobs: np.ndarray  # float32, same length as tokens
+    # vLLM's raw prompt_logprobs structure, kept only when the request asked for it.
+    prompt_logprobs: list[Any] | None = None
+
+
+def _floats_from_list(values: list[Any], expected_len: int) -> np.ndarray:
+    """Float32 array from a decoded list, filling absent values with zeros.
```
Comment (avigyabb) [inline skyrl/tinker/extra/completion_decode.py:38]: take a look
